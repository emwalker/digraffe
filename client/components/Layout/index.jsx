// @flow
import React from 'react'
import type { Node } from 'react'

type Props = {
  children?: Node,
}

const Layout = ({ children }: Props) => (
  <div>
    <nav className="navbar navbar-expand-lg navbar-light bg-light flex-column flex-md-row bd-navbar">
      <a className="navbar-brand mb-0 h1" href="/">Digraffe</a>
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <a className="nav-link test-topics-page" href="/topics">Topics</a>
        </li>
        <li className="nav-item">
          <a className="nav-link test-links-page" href="/links">Links</a>
        </li>
      </ul>
    </nav>
    <div className="container">
      { children }
    </div>
  </div>
)

Layout.defaultProps = {
  children: null,
}

export default Layout
