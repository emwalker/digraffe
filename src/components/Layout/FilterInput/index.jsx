// @flow
import React, { Component } from 'react'

type Props = {
  onEnter?: ?Function,
  value: string,
}

type State = {
  value: string,
}

class FilterInput extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      value: props.value,
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ value: nextProps.value })
  }

  onChange = (event) => {
    this.setState({ value: event.target.value })
  }

  onEnter = (string) => {
    if (this.props.onEnter)
      this.props.onEnter(string)
  }

  onKeyPress = (event) => {
    if (event.key === 'Enter')
      this.onEnter(event.target.value)
  }

  render = () => (
    <input
      aria-label="Filter input"
      className="form-control"
      onChange={this.onChange}
      onKeyPress={this.onKeyPress}
      placeholder="Search"
      size="40"
      type="text"
      value={this.state.value}
    />
  )
}

export default FilterInput
