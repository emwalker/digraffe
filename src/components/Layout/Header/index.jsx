// @flow
import React from 'react'
import MediaQuery from 'react-responsive'
import classNames from 'classnames'

import DesktopNav from './DesktopNav'
import MobileNav from './MobileNav'
import { header } from './styles.module.css'

type Props = {
  viewer: Object,
  defaultOrganization: Object,
}

export default ({ viewer, defaultOrganization }: Props) => {
  const classes = classNames(header, 'clearfix mb-3 d-flex px-md-6 px-lg-4')

  return (
    <div className={classes}>
      <div style={{ width: '100%' }}>
        <MediaQuery maxWidth={544}>
          <MobileNav viewer={viewer} />
        </MediaQuery>
        <MediaQuery minWidth={544}>
          <DesktopNav
            viewer={viewer}
            defaultOrganization={defaultOrganization}
          />
        </MediaQuery>
      </div>
    </div>
  )
}
