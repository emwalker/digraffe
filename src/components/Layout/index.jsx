// @flow
import React from 'react'
import type { Node } from 'react'
import { Link } from 'found'

type Props = {
  children?: Node,
}

const Layout = ({ children }: Props) => (
  <div>
    <header className="masthead">
      <div className="container">
        <a className="masthead-logo" href="/">
          Digraph
        </a>
        <nav className="masthead-nav">
          <Link
            className="test-topics-page"
            activeClassName="active"
            href="/topics"
            to="/topics"
          >
            Topics
          </Link>
          <Link
            className="test-links-page"
            activeClassName="active"
            href="/links"
            to="/links"
          >
            Links
          </Link>
        </nav>
      </div>
    </header>
    <div className="container">
      { children }
    </div>
    <div className="container">
      <footer className="my-6 pt-4 border-top">
        <p className="mb-2">
          Available for use under the MIT{' '}
          <a href="https://github.com/emwalker/digraph/blob/master/LICENSE.md">license</a>.
          Copyright Eric Walker 2018.
          Derived from Githubâ€™s <a href="https://primer.github.io/">Primer</a>.
        </p>
      </footer>
    </div>
  </div>
)

Layout.defaultProps = {
  children: null,
}

export default Layout
