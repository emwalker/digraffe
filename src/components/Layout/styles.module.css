.layout {
  min-height: 100vh;
  overflow: hidden;
  display: block;
  position: relative;
  padding-bottom: 140px;
}

.footer {
  position: absolute;
  width: 100%;
  bottom: 0;
}
