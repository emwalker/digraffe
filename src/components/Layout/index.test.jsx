import React from 'react'
import { shallow } from 'enzyme'
import Layout from './index'

describe('<Layout />', () => {
  const router = {
    push: jest.fn(),
  }

  const location = {
    query: { q: '' },
    search: '',
  }

  const view = {
    viewer: {},
  }

  const wrapper = shallow(
    <Layout
      router={router}
      location={location}
      view={view}
    >
      <div>some view</div>
    </Layout>,
  )

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('does not include links for the moment', () => {
    expect(wrapper.find('.test-links-page')).toHaveLength(0)
  })
})
