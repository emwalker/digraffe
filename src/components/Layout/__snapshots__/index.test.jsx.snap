// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders 1`] = `
<div
  className="layout"
>
  <div
    className="container"
  >
    <Header />
    <FlashMessages
      initialAlerts={Array []}
    />
    <div>
      some view
    </div>
  </div>
  <footer>
    <div
      className="container my-6 pt-2 border-top"
    >
      <p
        className="mb-2"
      >
        Available for use under the MIT
         
        <a
          href="https://github.com/emwalker/digraph/blob/master/LICENSE.md"
        >
          license
        </a>
        . Â© Eric Walker.
      </p>
    </div>
  </footer>
</div>
`;
